_dev/coding-style.md

```diff
@@ -6,8 +6,65 @@ description: The preferred style for code committed to NeoMutt
 
 # Coding Style
 
-Every project has its own coding style, this one is no exception.
-However, I haven't written down the rules.  For now, ...
+## at the moment (pre-fork)
 
-- Trailing whitespace is **evil**
+Generally speaking, we don't want to diverge much from upstream mutt. We're
+still merging patches in - and if we didn't follow upstream's coding style,
+merging would be much harder.
 
+That means:
+
+* put your the braces of functions in new lines, like so:
+
+```c
+struct q_class_t
+{
+  int length;
+  int index;
+  int color;
+  char *prefix;
+  struct q_class_t *next, *prev;
+  struct q_class_t *down, *up;
+};
+```
+
+* define functions with underscores like `void mutt_account_unsetpass (..)`
+
+* what a function does *should* be clear without the need to look into ites
+  implementation - that means use expressive names!
+
+* No global variable is a good variable - therefore avoid them.
+
+* Explain with comments what a function does. That includes:
+
+    - the return value and their meaning
+
+    - parameters and their meaning
+
+
+* Use only `/* Comment */` style comments. Don't use single line comments (`//
+  Comment`)
+
+* use a two spaces based indentation
+
+* Wrap your code and comments at 80 Columns.
+
+* Remove the braces of control statements (`if`, `switch` and such) which are
+  only one line big.
```

Comment:
1. language: *remove braces around control statement's blocks that contain only one statement
2. don't put `switch` as an example, if it has only one statement inside, then it should be an if ☺

Generally speaking, in programing languages a line has little meaning: you have *tokens* forming an *expression* being part of a *statement*.

As a mnemonic, in C, a statement always ends with a semicolon, and an expression is always something that can be assigned.